include/nlohmann/json.hpp

```diff
@@ -1458,6 +1553,78 @@ class basic_json
         assert_invariant();
     }
 
+    /*!
+    @brief explicitly create a binary array from an already constructed copy of
+    its base type
+
+    Creates a JSON binary array value from a given `binary_t`. Because binary
+    values are not part of the JSON specification, this is not the constructor
+    you want unless you specifically want to serialize to a binary format.
+
+    @note Note, this function exists because of the difficulty in correctly
```

Comment:
This comment is very defensive. I think it would be OK to mention that binary data is allowed in formats like CBOR, MessagePack, etc., and this constructor is meant to create binary values to be serialized using these formats.